src/mlpack/methods/ne/cne.hpp

```diff
@@ -0,0 +1,186 @@
+ /**
+ * @file cne.hpp
+ * @author Bang Liu
+ *
+ * Definition of CNE class.
+ */
+#ifndef MLPACK_METHODS_NE_CNE_HPP
+#define MLPACK_METHODS_NE_CNE_HPP
+
+#include <cstddef>
+#include <cstdio>
+
+#include <mlpack/core.hpp>
+
+#include "link_gene.hpp"
+#include "neuron_gene.hpp"
+#include "genome.hpp"
+#include "species.hpp"
+#include "tasks.hpp"
+#include "parameters.hpp"
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This class implements Conventional Neuro-evolution (CNE): weight
+ * evolution on topologically fixed neural networks.
+ */
+template<typename TaskType>
+class CNE {
+ public:
+  // Parametric constructor.
+  CNE(TaskType task, Genome& seedGenome, Parameters& params) {
```

Comment:
I agree, that for some methods you have to specify a bunch of parameters, so I think the parameter class comes in handy. On the other side, I like to provide a constructor that works right out of the box and starts with some reasonable parameters. I think in the case of the CNE method, we can specify such parameters. Maybe we can provide an constructor for such a situation, what do you think?